storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArray.java

```diff
@@ -695,9 +680,11 @@ private boolean hasAncestorAtSlot(
     return node != null && requiredRoot.equals(node.getBlockRoot());
   }
 
-  private boolean doesCheckpointEpochMatch(final Checkpoint actual, final Checkpoint required) {
-    return required.getEpoch().equals(initialEpoch)
-        || actual.getEpoch().equals(required.getEpoch());
+  private boolean isVotingSourceWithinAcceptableRange(
```

Comment:
in my version I was passing epochs to this function since this actually deals only with epochs (so we get them once and lines here are shorter)

Reply:
originally i was rationalising that change and needed checkpoint but i think its probably ok now, will look at a refactor here.